### PR TITLE
Cleanup fix

### PR DESCRIPTION
**Pull request by jnwatson**
Fix cleanup, suppress CancelledErrors, add unit tests.

### CODE REVIEW

File: synapse/lib/spawn.py [hunk 1]
```diff
@@ -254,7 +294,7 @@ def ctor(argv):
         self.onfini(self.prox.fini)
 
         self.hive = await s_hive.openurl(f'cell://{self.dirn}', name='*/hive')
-        self.onfini(self.hive.fini)
+        self.onfini(self.hive)
```
Comment (vEpiphyte) [inline synapse/lib/spawn.py]: what'd this change?

File: synapse/tests/utils.py [hunk 2]
```diff
@@ -1192,12 +1192,15 @@ def getAsyncLoggerStream(self, logname, mesg=''):
         handler = logging.StreamHandler(stream)
         slogger = logging.getLogger(logname)
         slogger.addHandler(handler)
+        level = slogger.level
+        slogger.setLevel('INFO')
```
Comment (vEpiphyte) [inline synapse/tests/utils.py]: If we're here - lets set the logger to DEBUG and make sure we also make the change to ``getLoggerStream`` ?

File: synapse/daemon.py [hunk 3]
```diff
@@ -168,13 +168,17 @@ class AsyncGenr(s_share.Share):
...
+                if not isinstance(context, asyncio.CancelledError):
+                    logger.error(f'Error during DmonSpawn call: {repr(context)}')
                 await link.fini()
             return
 
         except Exception as e:
 
-            logger.exception(f'error during task: {meth.__name__}')
+            if isinstance(e, asyncio.CancelledError):
+                logger.info('t2call task %s cancelled', meth.__name__)
```
Comment (vEpiphyte) [inline synapse/daemon.py]: fstring vs `%` string. we should pick one here.

File: synapse/tests/test_cortex_spawn.py [hunk 4]
```diff
@@ -91,10 +94,46 @@ class CoreSpawnTest(s_test.SynTest):
...
+                    self.false(1)
+
+                self.eq(donecount, n)
+
+                # test kill -9 ing a spawn proc
+
+                victimpid = core.spawnpool.spawnq[0].proc.pid
+                sig = signal.SIGKILL
+
+                async def taskfunc2():
```
Comment (vEpiphyte) [inline synapse/tests/test_cortex_spawn.py]: whats the test criteria here? that the test doesn't wedge?

File: synapse/tests/test_cortex_spawn.py [hunk 5]
```diff
@@ -91,10 +94,46 @@ class CoreSpawnTest(s_test.SynTest):
...
-                #msgs = await prox.storm('inet:ipv4:_woot=10', opts=opts).list()
-                #podes = [m[1] for m in msgs if m[0] == 'node']
-                #self.len(1, podes)
-                #self.eq(podes[0][0], ('inet:ipv4', 0x01020304))
+                await core.addFormProp('inet:ipv4', '_woot', ('int', {}), {})
+                await core.nodes('[inet:ipv4=1.2.3.4 :_woot=10]')
+                await core.view.layers[0].layrslab.waiter(1, 'commit').wait()
+                podes = await prox.eval('inet:ipv4=1.2.3.4', opts=opts).list()
+                self.len(1, podes)
+                self.eq(podes[0][1]['props'].get('_woot'), 10)
```
Comment (vEpiphyte) [inline synapse/tests/test_cortex_spawn.py]: what about lifting by model extensions?

**Comment by jnwatson** [comment]
It finis the hive before killing the active tasks.

**Comment by jnwatson** [comment]
ok

**Comment by jnwatson** [comment]
done for the whole file

**Comment by jnwatson** [comment]
yes

**Comment by jnwatson** [comment]
Good call. It doesn't work. Commented out and documented.

**Comment by vEpiphyte** [comment]
@jnwatson from your last commit

> epiphyte review. One test still fails

What test fails for you here? In CI tests are passing. Is something still incomplete here?

**Comment by jnwatson** [comment]
One test at the end of test_cortex_spawn.py is commented out. It fails. It
will continue to fail until core.getModelDefs works.

Nic

File: synapse/daemon.py [hunk 6]
```diff
@@ -197,12 +201,12 @@ class AsyncGenr(s_share.Share):
     except s_exc.DmonSpawn as e:
         context = e.__context__
         if context:
-            logger.error(f'Error during DmonSpawn call: {repr(context)}')
+            logger.error('Error during DmonSpawn call: %r', context)
```
Comment (invisig0th) [inline synapse/daemon.py]: Is there a reason we're heading back from f strings?

**Comment by jnwatson** [comment]
Logging has its own rules, mostly to avoid rendering if the log level is
set differently.

Nic

**Pull request merged**